### PR TITLE
Removed deprecated localhost_only

### PR DESCRIPTION
**Pull request by ahcorde**
This method `rcl_get_localhost_only(rmw_localhost_only_t * localhost_only);` was not deprecated, not sure if we can remove this.

This is deprecated at least since `iron`

### CODE REVIEW

**Comment by ahcorde** [comment]
friendly ping @fujitatomoya

**Comment by fujitatomoya** [comment]
@ahcorde sorry for being late, but this is under my radar. let me have a couple of days!

File: rcl/test/rcl/test_discovery_options.cpp [hunk 1]
```diff
@@ -293,29 +293,6 @@ TEST(TestDiscoveryInfo, test_with_localhost_only) {
   }
 
   {
-    // Only ROS_LOCALHOST_ONLY is enabled
```
Comment (fujitatomoya) [inline rcl/test/rcl/test_discovery_options.cpp:296]: I think we can also remove the comments on this test.

**Review by fujitatomoya — APPROVED** [review]
looks good to me as rcl, the following rmw implementation PRs to remove member variable aligns with this PR.

- https://github.com/ros2/rmw/pull/376
- https://github.com/ros2/rmw_implementation/pull/239
- https://github.com/ros2/rmw_fastrtps/pull/773
- https://github.com/ros2/rmw_connextdds/pull/156
- https://github.com/ros2/rmw_cyclonedds/pull/508

No documentation required to change, https://docs.ros.org/en/rolling/Tutorials/Advanced/Improved-Dynamic-Discovery.html already explains well.

**Comment by fujitatomoya** [comment]
@ahcorde can you take a look at related rmw PRs? just removing the `localhost_only` member from the structure, test needs to use discovery option instead.

**Comment by ahcorde** [comment]
Running full CI, included all PRs

Gist: https://gist.githubusercontent.com/ahcorde/3719f45d435e5d9adf75ea3d563f17fb/raw/05e68f5dade0fb9ef5e7bbd59cc7cec770351042/ros2.repos

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21565/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15853/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1063/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22285/)

**Pull request merged**